cmd/account/mgmt/account-unassign.go

```diff
@@ -131,54 +127,124 @@ func (o *accountUnassignOptions) run() error {
 	// loop through accounts list and untag and move them back into root OU
 	for _, id := range accountIdList {
 
+		// untag account
 		err = o.untagAccount(id)
 		if err != nil {
 			return err
 		}
 
+		// move account
 		err = o.moveAccount(id, rootID, destinationOU)
 		if err != nil {
 			return err
 		}
+		// instantiate new client with AssumeRole
+		newAWSClient, err := o.assumeRoleForAccount(id)
+		if err != nil {
+			return err
+		}
+		// delete roles
+		err = o.deleteRoles(newAWSClient)
+		if err != nil {
+			return err
+		}
+		// list iam users created by each account and append to slice
+		users, err := o.listUsersFromAccount(newAWSClient, id)
+		if err != nil {
+			return err
+		}
+
+		allUsers = append(allUsers, users...)
+
 	}
 
-	// Delete login profile
-	err = o.deleteLoginProfile(accountUsername)
-	if err != nil {
-		fmt.Println(err)
+	for _, userName := range allUsers {
+
+		// Delete login profile
+		err = o.deleteLoginProfile(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete access keys
+		err = o.deleteAccessKeys(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete signing certificates
+		err = o.deleteSigningCert(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete policies
+		err = o.deletePolicies(userName)
```

Comment:
I'm still not convinced that we need the username for these parameters.  We should be deleting _all_ user created policies, all user created login profiles, all users, all user-created roles, etc.  So each one of these functions would be something like:

```
func o.deleteUsers() error {
    all_users := aws.ListUsers()
    for user in all_users {
        aws.DeleteUser(user.Arn)
    }
}

func o.deleteRoles() error {
    all_roles := aws.ListRoles()
    for role in all_roles {
        if role.Name == "OrganizationAccountAccessRole" {
            continue
        }
        aws.DeleteRole(role.Arn)
    }
}
```

So each of these functions would be a self-contained "list all" and then "remove if not managed by AWS".

Reply:
all the delete operations are inside a loop that iterates through the list of IAM users created by the user we wanna clean. Instead of doing listUsers -> delete -> listUsers -> delete ...... i did for each user in listUsers -> { perform all the deletions}

Reply:
Ooh, I see, this is deleting all of the policies that are created as UserPolicies, but I don't think this also deletes standalone policies.

So in the AWS console, if you go to the Users page and create a policy under a user it will attach inline policies to that user.  These are UserPolicies.  If you go into the Policies page and create a new Policy, this is just a Policy, I believe.

Either way, I'll have to do more research to make sure that my assumption above is correct, because my brain tends to instantly brain-dump IAM stuff as soon as I don't directly need it anymore, I'm sure that's a coping mechanism 😂. With that said, I'll give this PR a test as is and we can clean up lingering IAM standalone policies if they're not getting cleaned up as a separate body of work.